code/modules/atmospherics/machinery/datum_pipeline.dm

```diff
@@ -154,6 +154,11 @@
 	var/partial_heat_capacity = total_heat_capacity*(share_volume/air.volume)
 	var/target_temperature
 	var/target_heat_capacity
+	// first calculate heat from radiation. there's an implied "* 1 tick" here.
+	// 0.05 magic multiplicand is, first, 0.1 deciseconds; second, half of the radiation's going right back into the gas.
+	var/share_constant = STEFANBOLTZMANN*(share_volume**(2/3))*0.05
+	// Minimizing temp to 4 billion is mostly to prevent -infinity temperatures.
+	var/heat = share_constant*(min(air.temperature,4000000000)**4)
```

Comment:
I think these should be static vars, but the PR has been open for a while and I entrust there is nothing wrong with your knowledge of heat radiation or osmosis.